File: frontend/graylink-web/src/api/emby.ts
import { api } from '@/utils/request'
import type { EmbyLibrary, EmbyStatus } from '@/types/emby'
import type { ApiResponse } from '@/types/response'

export const embyApi = {
  // 检查服务器状态
  checkStatus() {
    return api.get<ApiResponse<EmbyStatus>>('/emby/status')
  },

  // 获取媒体库列表
  getLibraries() {
    return api.get<ApiResponse<EmbyLibrary[]>>('/emby/libraries')
  },

  // 按路径刷新媒体库
  refreshByPaths(paths: string[]) {
    return api.post<ApiResponse<void>>('/emby/refresh', { paths })
  },

  // 刷新整个媒体库
  refreshRoot() {
    return api.post<ApiResponse<void>>('/emby/refresh/root')
  },

  // 测试连接
  test() {
    return api.post<ApiResponse<void>>('/emby/test')
  },

  // 刷新媒体库
  refreshLibrary() {
    return api.post<ApiResponse<void>>('/emby/refresh')
  }
}

File: backend/app/handlers/emby.py
from fastapi import APIRouter, HTTPException
from typing import List, Optional
from pydantic import BaseModel
from app.utils.emby import EmbyClient
from app.utils.config import get_config

router = APIRouter(tags=["emby"])

class EmbyStatus(BaseModel):
    connected: bool
    version: Optional[str] = None
    server_name: Optional[str] = None

class EmbyLibrary(BaseModel):
    id: str
    name: str
    path: str

@router.get("/status")
async def check_status():
    return {
        "code": 0,
        "data": {
            "connected": True,
            "version": "4.7.0.0",
            "server_name": "Emby Server"
        }
    }

@router.get("/libraries")
async def get_libraries():
    return {
        "code": 0,
        "data": []
    }

@router.post("/refresh")
async def refresh_by_paths(paths: List[str]):
    return {
        "code": 0,
        "data": None
    }

@router.post("/refresh/root")
async def refresh_root():
    return {
        "code": 0,
        "data": None
    }

@router.post("/test")
async def test_connection():
    return {
        "code": 0,
        "data": None
    } 